From: Michal Hocko <mhocko@suse.cz> 
Subject: [PATCH] mm, kabi: support madvise(MADV_FREE)
Patch-mainline: never (suse specific)
References: fate#321375

PGLAZYFREED changes offset of all counters which are behind which can confuse
out of tree drivers which are using static inline helpers to update and
retrieve those counters. Let's move it to the end of vm_event_item so that this
code is not affected. PGLAZYFREED is only used from the core kernel which will
get recompiled and so see no issues. Increased NR_VM_EVENT_ITEMS is also OK for
the out-of-tree modules because they will always allocate the previously
memory. The per-cpu layout will change but this will get resolved at the link
time (aka module load time for external modules) so this is OK as well.

pglazyfreed has to match the vm_event_item so move it as well.

As this would still trigger the kabi warnings hide them.

Signed-off-by: Michal Hocko <mhocko@suse.cz>

---
 include/linux/vm_event_item.h |    4 +++-
 mm/vmstat.c                   |    4 +++-
 2 files changed, 6 insertions(+), 2 deletions(-)

--- a/include/linux/vm_event_item.h
+++ b/include/linux/vm_event_item.h
@@ -25,7 +25,6 @@ enum vm_event_item { PGPGIN, PGPGOUT, PS
 		FOR_ALL_ZONES(PGALLOC),
 		PGFREE, PGACTIVATE, PGDEACTIVATE,
 		PGFAULT, PGMAJFAULT,
-		PGLAZYFREED,
 		FOR_ALL_ZONES(PGREFILL),
 		FOR_ALL_ZONES(PGSTEAL_KSWAPD),
 		FOR_ALL_ZONES(PGSTEAL_DIRECT),
@@ -94,6 +93,9 @@ enum vm_event_item { PGPGIN, PGPGOUT, PS
 		VMACACHE_FIND_HITS,
 		VMACACHE_FULL_FLUSHES,
 #endif
+#ifndef __GENKSYMS__
+		PGLAZYFREED,
+#endif
 		NR_VM_EVENT_ITEMS
 };
 
--- a/mm/vmstat.c
+++ b/mm/vmstat.c
@@ -758,7 +758,6 @@ const char * const vmstat_text[] = {
 
 	"pgfault",
 	"pgmajfault",
-	"pglazyfreed",
 
 	TEXTS_FOR_ZONES("pgrefill")
 	TEXTS_FOR_ZONES("pgsteal_kswapd")
@@ -847,6 +846,9 @@ const char * const vmstat_text[] = {
 	"vmacache_full_flushes",
 #endif
 #endif /* CONFIG_VM_EVENTS_COUNTERS */
+#ifndef __GENKSYMS__
+	"pglazyfreed",
+#endif
 };
 #endif /* CONFIG_PROC_FS || CONFIG_SYSFS || CONFIG_NUMA */
 
